opm/common/utility/CSRGraphFromCoordinates.hpp

```diff
@@ -0,0 +1,490 @@
+
+    public:
+        /// Clear all internal buffers, but preserve allocated capacity.
+        void clear();
+
+        /// Add flow rate connection between regions.
+        ///
+        /// \param[in] v1 First vertex in vertex pair.  Used as row index.
+        ///
+        /// \param[in] r2 Second vertex in vertex pair.  Used as column index.
+        ///
+        /// If both vertex IDs are the same then this function does nothing.
+        void addConnection(VertexID v1, VertexID v2);
+
+        /// Form CSR adjacency matrix representation of input graph from
+        /// connections established in previous calls to addConnection().
+        ///
+        /// \param[in] maxNumVertices Number of rows in resulting CSR
+        ///     matrix.  If prior calls to addConnection() supply vertex IDs
+        ///     (row indices) greater than or equal to \p maxNumVertices,
+        ///     then method compress() will throw \code
+        ///     std::invalid_argument \endcode.
+        void compress(Offset maxNumVertices);
+
+        /// Retrieve number of rows (source entities) in input graph.
+        /// Corresponds to value of argument passed to compress().  Valid
+        /// only after calling compress().
+        Offset numVertices() const;
+
+        /// Retrieve number of edges (non-zero matrix elements) in input
+        /// graph.
+        Offset numEdges() const;
+
+        /// Read-only access to compressed structure's start pointers.
+        const Start& startPointers() const
+        {
+            return this->csr_.startPointers();
+        }
+
+        /// Read-only access to compressed structure's column indices,
+        /// ascendingly sorted per row.
+        const Neighbours& columnIndices() const
+        {
+            return this->csr_.columnIndices();
+        }
+
+        /// Read-only access to mapping from input order vertex pairs to
+        /// compressed structure's edge index (location in ja/sa).
+        ///
+        /// Available only if client code sets TrackCompressedIdx=true.
+        /// Compiler diagnostic, typically referring to 'enable_if', if
+        /// client code tries to call this function without setting
+        /// class parameter TrackCompressedIdx=true.
+        template <typename Ret = const Start&>
+        std::enable_if_t<TrackCompressedIdx, Ret> compressedIndexMap() const
+        {
+            return this->csr_.compressedIndexMap();
+        }
+
+        // MessageBufferType API should be similar to Dune::MessageBufferIF
+        template <class MessageBufferType>
+        void write(MessageBufferType& buffer) const
+        {
+            this->csr_.write(buffer);
+        }
+
+        // MessageBufferType API should be similar to Dune::MessageBufferIF
+        template <class MessageBufferType>
+        void read(MessageBufferType& buffer)
+        {
+            auto other = CSR{};
+            other.read(buffer);
+
+            this->uncompressed_
```

Comment:
Can we not just keep the uncompressed_ part empty here? Can we call `csr_.read(buffer)` instead?